src/test/java/org/prebid/server/floors/PriceFloorsConfigResolverTest.java

```diff
@@ -0,0 +1,256 @@
+        verify(metrics).updateAlertsConfigFailed("some-id", MetricName.price_floors);
+    }
+
+    @Test
+    public void updateFloorsConfigShouldReturnDefaultConfigIfMaxRulesMoreThanMaximumValue() {
+        // when
+        final Future<?> future = testingInstance.updateFloorsConfig(
+                accountWithFloorsFetchConfig(config -> config.maxRules(Integer.MAX_VALUE + 1L)));
+
+        // then
+        assertThat(future.succeeded()).isTrue();
+        assertThat(future.result())
+                .isEqualTo(withDefaultFloorsConfig().toBuilder().id("some-id").build());
+        verify(metrics).updateAlertsConfigFailed("some-id", MetricName.price_floors);
+    }
+
+    @Test
+    public void updateFloorsConfigShouldReturnDefaultConfigIfMaxFileSizeLessThanMinimumValue() {
+        // when
+        final Future<?> future = testingInstance.updateFloorsConfig(
+                accountWithFloorsFetchConfig(config -> config.maxFileSize(-1L)));
+
+        // then
+        assertThat(future.succeeded()).isTrue();
+        assertThat(future.result())
+                .isEqualTo(withDefaultFloorsConfig().toBuilder().id("some-id").build());
+        verify(metrics).updateAlertsConfigFailed("some-id", MetricName.price_floors);
+    }
+
+    @Test
+    public void updateFloorsConfigShouldReturnDefaultConfigIfMaxFileSizeMoreThanMaximumValue() {
+        // when
+        final Future<?> future = testingInstance.updateFloorsConfig(
+                accountWithFloorsFetchConfig(config -> config.maxFileSize(Integer.MAX_VALUE + 1L)));
+
+        // then
+        assertThat(future.succeeded()).isTrue();
+        assertThat(future.result())
+                .isEqualTo(withDefaultFloorsConfig().toBuilder().id("some-id").build());
+        verify(metrics).updateAlertsConfigFailed("some-id", MetricName.price_floors);
+    }
+
+    private static Account accountWithFloorsFetchConfig(
+            UnaryOperator<AccountPriceFloorsFetchConfig.AccountPriceFloorsFetchConfigBuilder> configCustomizer) {
+        return Account.builder()
+                .id("some-id")
+                .auction(AccountAuctionConfig.builder()
+                        .priceFloors(AccountPriceFloorsConfig.builder()
+                                .enforceFloorsRate(10)
+                                .fetch(configCustomizer.apply(
+                                        AccountPriceFloorsFetchConfig.builder()
+                                                .maxAgeSec(1000L)
+                                                .periodSec(600L)
+                                                .timeout(100L)
+                                                .maxRules(100L)
+                                                .maxFileSize(100L)
+                                ).build())
+                                .build())
+                        .build())
+                .build();
+    }
+
+    private static Account withDefaultFloorsConfig() {
```

Comment:
We are modifying Account so UnaryOperator would be nice to have (at least I see usages of it in tests here)